subsys/logging/log_backend_rtt.c

```diff
@@ -232,14 +231,14 @@ static void log_backend_rtt_init(void)
 	}
 
 	host_present = true;
-	panic_mode = 0;
+	sync_mode = IS_ENABLED(CONFIG_LOG_IMMEDIATE) ? true : false;
```

Comment:
mutex was used in case of LOG_IMMEDIATE. That probably could trigger rescheduling inside synchronous operation. Not sure how that was handled by kernel since interrupts where locked when this function was called.